core/imagefs_linux.go

```diff
@@ -20,31 +20,31 @@ limitations under the License.
 package core
 
 import (
-	"context"
-	"os"
-	"path/filepath"
+	"syscall"
 	"time"
 
 	"github.com/sirupsen/logrus"
 
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
-// ImageFsInfo returns information of the filesystem that is used to store images.
-func (ds *dockerService) ImageFsInfo(
-	_ context.Context,
-	_ *runtimeapi.ImageFsInfoRequest,
-) (*runtimeapi.ImageFsInfoResponse, error) {
+// ImageFsInfo returns information of the filesystem of docker data root.
+func (ds *dockerService) imageFsInfo() (*runtimeapi.ImageFsInfoResponse, error) {
 	info, err := ds.getDockerInfo()
 	if err != nil {
 		logrus.Error(err, "Failed to get docker info")
 		return nil, err
 	}
 
-	bytes, inodes, err := dirSize(filepath.Join(info.DockerRootDir, "image"))
```

Comment:
This additional work looks promising to me (need to spend more time looking at specifics later in the week when I find time); it's worth noting that this will conflict with/obsolete https://github.com/Mirantis/cri-dockerd/pull/221.

cc @surik